packages/dev/docs/src/docs.js

```diff
@@ -36,29 +37,60 @@ for (let link of links) {
         <Pressable>
           {/* eslint-disable jsx-a11y/click-events-have-key-events */}
           {/* eslint-disable jsx-a11y/anchor-is-valid */}
-          <a role="link" tabIndex={0} data-link={link.dataset.link} className={link.className} onClick={e => e.preventDefault()}>{link.textContent}</a>
+          <a role="button" tabIndex={0} aria-haspopup="dialog" data-link={link.dataset.link} className={link.className} onClick={e => e.preventDefault()}>{link.textContent}</a>
         </Pressable>
         <LinkPopover id={link.dataset.link} />
       </DialogTrigger>
     </ThemeProvider>
   , container);
 
-  link.parentNode.replaceChild(container, link);
+  link.parentNode.replaceChild(container.firstElementChild, link);
 }
 
 function LinkPopover({id}) {
   let ref = useRef();
   let [breadcrumbs, setBreadcrumbs] = useState([document.getElementById(id)]);
 
+  const onBlurCurrentBreadcrumb = useCallback((event) => {
+    event.target.removeEventListener('blur', onBlurCurrentBreadcrumb);
+    event.target.tabIndex = -1;
+  },
+  []);
+
+  const isCancelledRef = useRef(false); 
+  const timeoutRef = useRef();
   useEffect(() => {
     let links = ref.current.querySelectorAll('[data-link]');
     for (let link of links) {
+      // Links with [data-link] will open within the LinkPopover, so [aria-haspopup] is not appropriate.
+      link.removeAttribute('aria-haspopup');
+      // Add click event handler so that the link updates the content of the popover.
       link.addEventListener('click', (e) => {
         e.preventDefault();
         setBreadcrumbs([...breadcrumbs, document.getElementById(link.dataset.link)]);
       });
     }
-  }, [breadcrumbs]);
+
+    if (isCancelledRef.current) {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+      return;
+    }
+
+    // Wait a frame, then focus the current breadcrumb for the popover, 
+    // which ensures that focus is not lost to the document.body when the LinkPopover content changes.
+    timeoutRef.current = setTimeout(() => {
+      const currentBreadcrumb = ref.current.closest(`.${docsStyle.popover}`).querySelector('[aria-current]');
+      if (currentBreadcrumb) {
+        currentBreadcrumb.tabIndex = 0;
+        currentBreadcrumb.addEventListener('blur', onBlurCurrentBreadcrumb);
+        focusWithoutScrolling(currentBreadcrumb); 
+      }
+    }, 0);
+
+    () => isCancelledRef.current = true;
```

Comment:
i still do not follow the logic going on here
is this really two useEffects?

or a useLayoutEffect and a useEffect?

at the very least, this return needs to actually cancel the setTimeout so it can't be called on an unmounted component where it can't access anything because it's no longer in the dom

Reply:
Ok, I think I have it now: https://github.com/adobe/react-spectrum/pull/728/commits/6526bb0f113a05366cfbaeb648ee56b928ecb5f3#diff-27e06259cf9af05851fc3cc84ecc75120dcbf242051ad4acc6bfa09d892f8b07R61-R96